src/storage/backup/ob_backup_iterator.cpp

```diff
@@ -262,6 +262,7 @@ int ObIBackupIndexIterator::read_data_file_trailer_(const ObBackupPath &backup_p
     ret = OB_ALLOCATE_MEMORY_FAILED;
     LOG_WARN("failed to alloc memory", K(ret), K(trailer_len));
   } else if (OB_FAIL(pread_file_(backup_path.get_obstr(), storage_info, mod, file_length - trailer_len, trailer_len, buf))) {
+    ret = OB_ERR_UNEXPECTED;
```

Comment:
`OB_FAIL` macro actually sets `ret`, which doesn't seem to be the effect here.

```
#define OB_FAIL(statement) (OB_UNLIKELY(::oceanbase::common::OB_SUCCESS != (ret = (statement))))
```

Reply:
Well, it's indeed not that simple a question, I'll try debugging it.